modules/insights/private-link-scope/tests/e2e/defaults/main.test.bicep

```diff
@@ -45,5 +45,9 @@ module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem'
   params: {
     enableDefaultTelemetry: enableDefaultTelemetry
     name: '${namePrefix}${serviceShort}001'
+    accessModeSettings: {
```

Comment:
Hey @johnlokerse, 
A great contribution as always. One question though: Would you think it makes sense to use this value for `accessModeSettings` as a default - i.e. making this parameter optional, or should it rather be mandatory as it's very different from case-to-case.

Reply:
We can set the default value for the required parameters. I think the default value is 'Open', but have to double-check this to make sure.

Reply:
It is the default. Based on this statement one could however also argue the opposite would be the better option: 
![image](https://github.com/Azure/ResourceModules/assets/5365358/b6ca78c1-1705-40ae-8af1-d8954afc748d)

Personally, I don't know the resource well enough to make an informed statement. It's always a balancing act in between security & usability.

Reply:
I would vote to have the `ingestionAccessMode` and `queryAccessMode` become optional and follow the ARM reference [documentation](https://learn.microsoft.com/en-us/azure/templates/microsoft.insights/privatelinkscopes?pivots=deployment-language-bicep#accessmodesettings). We could provide additional context into the description field like we do with other params around security recommendations. My preference would be to set them to `privateOnly` them if `privateEndpoints` are enabled like we do for other resources, but not sure if that's achievable in UDT.

Reply:
@AlexanderSehr @ahmadabdalla Happy new year! I was on holiday leave and going to pick this up again. I agree to make it optional and give it the default value `PrivateOnly`.

> My preference would be to set them to privateOnly them if privateEndpoints are enabled like we do for other resources, but not sure if that's achievable in UDT.

Wil check if this is possible.

Reply:
@ahmadabdalla It does not work to set the access mode to `PrivateOnly` via the UDT, but it can be achieved this way:

```bicep
@description('''
  Required. Specifies the access mode of ingestion or queries through associated private endpoints in scope. For security reasons, it is recommended to use PrivateOnly whenever possible to avoid data exfiltration.

  * Private Only (default) - Allows the connected virtual network to reach only Private Link resources. This mode is the most secure.
  * Open - Allows the connected virtual network to reach both Private Link resources and the resources not in the AMPLS resource. Data exfiltration cannot be prevented in this mode.
''')
param accessModeSettings accessModeType = {
  ingestionAccessMode: empty(privateEndpoints) ? 'Open' : 'PrivateOnly'
  queryAccessMode: empty(privateEndpoints) ? 'Open' : 'PrivateOnly'
}

@description('Optional. Configuration details for private endpoints. For security reasons, it is recommended to use private endpoints whenever possible.')
param privateEndpoints privateEndpointType
```

I have tested it in both scenario's:
* With an empty `privateEndpoints` parameter it sets access mode to 'Open'
* With a filled `privateEndpoints` parameter it sets access mode to 'PrivateOnly'

This way the accessModeSettings parameter is optional, but it adds a layer of complexity. I was thinking since we set the access mode programmatically maybe we should set is as a variable instead of a parameter.

What do you guys think @AlexanderSehr @ahmadabdalla ?

Reply:
Hey @johnlokerse, the issue with 'only'  a variable is that the user wouldn't have a way to overwrite the default behavior. In that sense I think it would always be good to have at least the option. That said, the logic does not have to be part of the parameter but could also be implemented where it is used. E.g.

```bicep
param accessModeSettings accessModeType

resource privateLinkScope 'microsoft.insights/privateLinkScopes@2021-07-01-preview' = {
  name: name
  location: location
  tags: tags
  properties: {
    accessModeSettings: accessModeSettings ?? {
      ingestionAccessMode: empty(privateEndpoints) ? 'Open' : 'PrivateOnly'
      queryAccessMode: empty(privateEndpoints) ? 'Open' : 'PrivateOnly'
    }
  }
}
```

Whether that's 'better' is debatable. The only point I'm trying to make is that we should keep the parameter as an option no matter what.

Reply:
Ah, yes I agree. Let's go for the option you're showing.

Reply:
Changes are pushed.